storm-core/src/clj/backtype/storm/daemon/worker.clj

```diff
@@ -288,7 +288,18 @@
                                       (filter-key (complement (-> worker :task-ids set))))
               needed-connections (-> needed-assignment vals set)
               needed-tasks (-> needed-assignment keys)
-              
+
+              closed-connections (into [] (for [[node+port socket] @(:cached-node+port->socket worker)]
+                                            (if (.isClosed socket) node+port)))
```

Comment:
-1 

We need take attention to unneeded connections but not closed connections ,
sometimes closed connections need to be reconnected by [retry policy](https://github.com/apache/storm/blob/master/storm-core/src/jvm/backtype/storm/messaging/netty/Client.java#L138) to make sure reconnect losted connections.
[refresh-connections](https://github.com/apache/storm/blob/master/storm-core/src/clj/backtype/storm/daemon/worker.clj#L313) will try remove unneeded connections

by the way:
 
```  
java.lang.IllegalArgumentException: No matching field found: isClosed for class  backtype.storm.messaging.netty.Client 
```
May be you should use [client.status()](https://github.com/apache/storm/blob/master/storm-core/src/jvm/backtype/storm/messaging/netty/Client.java#L191) somewhere when you need .